### PR TITLE
Issue 11789 allow receive from to take a netmask

### PR DESCRIPTION
**Pull request by nollymar**

### CODE REVIEW

**Comment by austindlawless** [comment]
I'm so excited.

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/publisher/endpoint/business/PublishingEndPointAPITest.java [hunk 1]
```diff
@@ -184,4 +185,51 @@ public void test() throws DotDataException {
...
+
+		PublishingEndPoint
+			endPoint =
+			createPublishingEndPoint("10", "G03", "Netmask Endpoint", "192.168.1.0/24",
+				"90", "http", true, "123456",
+				true);
+		// Insert test end point netmask
+		api.saveEndPoint(endPoint);
+
+		assertNotNull(api.findEnabledSendingEndPointByAddress("192.168.1.60"));
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/endpoint/business/PublishingEndPointAPITest.java]: we can assert a bit more, check if ids of endpoints are the same

File: dotCMS/src/integration-test/java/com/dotcms/publisher/endpoint/business/PublishingEndPointAPITest.java [hunk 2]
```diff
@@ -184,4 +185,51 @@ public void test() throws DotDataException {
...
+
+		// Delete endpoint
+		api.deleteEndPointById(endPoint.getId());
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/endpoint/business/PublishingEndPointAPITest.java]: this needs to happen inside a finally block, what if the previous asserts fail?

File: dotCMS/src/integration-test/java/com/dotcms/publisher/endpoint/business/PublishingEndPointAPITest.java [hunk 3]
```diff
@@ -184,4 +185,51 @@ public void test() throws DotDataException {
...
+
+	@Test
+	public void getEnabledSendingEndPointByAddress_returnEndpoint_whenIPAddressIsUsed() throws DotDataException {
+		PublishingEndPoint
+			endPoint =
+			createPublishingEndPoint("11", "G03", "IP Endpoint", "192.168.1.60",
+				"90", "http", true, "123456", true);
+		// Insert test end point netmask
+		api.saveEndPoint(endPoint);
+		assertNotNull(api.findEnabledSendingEndPointByAddress("192.168.1.60"));
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/endpoint/business/PublishingEndPointAPITest.java]: same as above

File: dotCMS/src/integration-test/java/com/dotcms/publisher/endpoint/business/PublishingEndPointAPITest.java [hunk 4]
```diff
@@ -184,4 +185,51 @@ public void test() throws DotDataException {
...
+
+		// Delete endpoint
+		api.deleteEndPointById(endPoint.getId());
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/endpoint/business/PublishingEndPointAPITest.java]: same as above

File: dotCMS/src/integration-test/java/com/dotcms/publisher/endpoint/business/PublishingEndPointAPITest.java [hunk 5]
```diff
@@ -184,4 +185,51 @@ public void test() throws DotDataException {
...
+			createPublishingEndPoint("12", "G03", "Netmask Endpoint", "192.168.1.0/24",
+				"90", "http", true, "123456",
+				true);
+		// Insert test end point netmask
+		api.saveEndPoint(endPoint);
+
+		assertNull(api.findEnabledSendingEndPointByAddress("192.168.2.60"));
+
+		// Delete endpoint
+		api.deleteEndPointById(endPoint.getId());
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/endpoint/business/PublishingEndPointAPITest.java]: same as above

File: dotCMS/src/main/java/com/dotcms/publisher/endpoint/business/PublishingEndPointFactoryImpl.java [hunk 6]
```diff
@@ -135,8 +136,16 @@ public void deleteEndPointById(String id) throws DotDataException {
...
+		boolean match;
 		for(PublishingEndPoint endPoint : allEndPoints) {
-			if(isMatchingEndpoint(endPoint.getAddress(), address) && endPoint.isEnabled() && endPoint.isSending())
+			ipOrNetMask = endPoint.getAddress();
+			if (ipOrNetMask.contains("/")) {
+				match = new SubnetUtils(ipOrNetMask).getInfo().isInRange(address);
+			} else {
+				match = isMatchingEndpoint(ipOrNetMask, address);
+			}
+			if(match && endPoint.isEnabled() && endPoint.isSending())
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/publisher/endpoint/business/PublishingEndPointFactoryImpl.java]: We could extract a bit of this, for example create a "boolean isInRange(address, netmask) method" that checks if the netmask is actually a netmask and in that case do the range check. That way we move some of the responsibility of this method to a new method

**Review by dsilvam — CHANGES_REQUESTED** [review]

**Comment by nollymar** [comment]
Code enhancements were committed

**Review by dsilvam — APPROVED** [review]

**Pull request merged**